cmake/Modules/LibAddPlugin.cmake

```diff
@@ -394,7 +400,20 @@ function (add_plugin PLUGIN_SHORT_NAME)
 	# message (STATUS "system incl are: ${ARG_INCLUDE_SYSTEM_DIRECTORIES}")
 	# message (STATUS "current bin ${CMAKE_CURRENT_BINARY_DIR}")
 
-	message (STATUS "Include Plugin ${PLUGIN_SHORT_NAME}")
+	if (ARG_ONLY_SHARED AND NOT BUILD_SHARED)
+		remove_plugin ("${PLUGIN_SHORT_NAME}"
+			       "it can only be built if BUILD_SHARED is enabled, both BUILD_FULL or BUILD_STATIC may be enabled as well")
+		return ()
+	endif (ARG_ONLY_SHARED AND NOT BUILD_SHARED)
+
+	set (STATUS_MESSAGE "Include Plugin ${PLUGIN_SHORT_NAME}")
+	if (ARG_ONLY_SHARED)
+
+		# also add it to the list of ONLY_SHARED plugins for exportsymbols.c configuration
+		set (STATUS_MESSAGE "${STATUS_MESSAGE} for shared builds")
+		set_property (GLOBAL APPEND PROPERTY SHARED_ONLY_PLUGINS "${PLUGIN_SHORT_NAME}")
```

Comment:
Please document what SHARED_ONLY_PLUGINS are for. (For example, see cmake/ElektraCache.cmake line 16)